### DISCUSSION TITLE
commit-callback script

### DISCUSSION THREAD

**Discussion by DanielK2013** [newren/git-filter-repo#202] [Ideas]
Would it be possible to read the commit-callback from a script file?
I've a very large commit-callback causing an arguments too long problem when executing.

**Comment by devkoa** [comment]
Yes, if i understand you correctly.
You can simply create a shell script to call the git-filter-repo.py

**Reply by DanielK2013** [reply]
i mean can i run
./git-filter-repo.py --commit-callback callback-script.py ??
Thanks for your reply

**Reply by devkoa** [reply]
I don't really think this could work. But instead you can do:
create a shell script and inside would look like (depends on your OS, i have 2 examples):
Linux OS:
/usr/bin/python3 git-filter-repo.py --commit-callback "arguments arguments arguments....n"
Windows OS:
C:/path/to/python/python.exe git-filter-repo.py --commit-callback "arguments arguments arguments....n"
And simply call the shell script:
./my_new_script.sh

**Comment by newren** [comment]
Isn't this basically what all the examples in contrib/filter-repo-demos/ do?  They all have callbacks (sometimes one, sometimes multiple callbacks) that are too big to be a one-liner and thus are stored in a script.

**Comment by newren** [comment]
Implemented by Shezan Baig in commit 5256c99e496b (Allow callback body to be loaded from a file, 2021-04-27)
